abuild.in

```diff
@@ -1125,7 +1136,7 @@ archcheck() {
 		[ "${subpkgarch:-$pkgarch}" != "noarch" ] && return 0
 		error "Arch specific binaries found so arch must not be set to \"noarch\""
 		return 1
-	elif [ "${subpkgarch:-$pkgarch}" != "noarch" ] && ! is_dev_pkg; then
+	elif [ "${subpkgarch:-$pkgarch}" != "noarch" ] && ! is_dev_pkg && ! is_static_pkg "$name"; then
```

Comment:
`archcheck()` doesn't define `$name` the same way `postcheck()` does.
Based on `is_dev_pkg` we should be using `$subpkgname`.
```suggestion
	elif [ "${subpkgarch:-$pkgarch}" != "noarch" ] && ! is_dev_pkg && ! is_static_pkg "$subpkgname"; then
```